backend-rust/src/transaction_event/smart_contracts.rs

```diff
@@ -58,12 +58,59 @@ pub struct ContractInitialized {
     pub amount:            Amount,
     pub init_name:         String,
     pub version:           ContractVersion,
+    pub input_parameter:   Vec<u8>,
     // All logged events by the smart contract during the transaction execution.
     pub contract_logs_raw: Vec<Vec<u8>>,
 }
 
 #[ComplexObject]
 impl ContractInitialized {
+    async fn message_as_hex(&self) -> ApiResult<String> { Ok(hex::encode(&self.input_parameter)) }
```

Comment:
Looking at the interfaces of these two functions, I would expect message_as_hex to be the simple hex upon the message. It is not why?

Cant we just make input param be optional as well, and then call message here?

Reply:
I made it input parameter optional as suggested. We discussed `message_as_hex` and `message` function and how they are used in a meeting.

Reply:
![Screenshot from 2025-03-27 09-12-31](https://github.com/user-attachments/assets/15326545-4773-46e0-8198-b1fa384ae13b)